src/_coustomware/socketMiddleware.js

```diff
@@ -0,0 +1,102 @@
+import { userActions } from '../_actions';
+import { config } from '../_helpers';
+import { userConstants } from '../_constants';
+
+const socketMiddleware = (function () {
+
+  var socket = null;
+
+  const onOpen = (ws, store, token) => evt => {
+    //Send a handshake, or authenticate with remote end
+    socket.send(token);
+    
+    //store.dispatch(actions.connected());
+  }
+
+  const onClose = (ws, store) => evt => {
+    //Tell the store we've disconnected
+    //store.dispatch(actions.disconnected());
+  }
+
+  const onMessage = (ws, store) => evt => {
+    //Parse the JSON message received on the websocket
```

Comment:
You should be switching on `msg.sub` and `msg.op`. This may work for now, but both add temp and add device use an op of 1 (which means add)

Reply:
Sure, added the fix. I forgot that we are using web-sockets to update the device and alarm also.